### DISCUSSION TITLE
clear or reset Map Store

### DISCUSSION THREAD

**Discussion by martinrojas** [ai/nanostores#212] [General]
Not sure if this fall under examples or actually a new feature, but for a Map store, what would be the correct way to clear all items? 

Taking the example of a shopping cart for a Map store, one of the most common features would be to clear the cart. The only way I see to do that now would be to write a loop that would read each item and mark it as `undefined` That would cause a lot of rendering and on a large enough store, cause significant performance problems

**Comment by ai** [comment]
Why not just `map.set({})` (if all keys allows `undefined`)?
